### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by olya-yevsieienko**

### CODE REVIEW

**Comment by Bohdan-Pyschchyk** [comment]
Well done🔥

**Review by Bohdan-Pyschchyk — APPROVED** [review]

File: src/averageAges.js [hunk 1]
```diff
@@ -3,59 +3,74 @@
...
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => onlyWithSon
+    ? people.some(
+      child => child.name === person.mother
+    ) && person.sex === 'm'
+    : people.some(
+      child => child.name === person.mother)
+  );
```
Comment (Olexandr6) [inline src/averageAges.js:61]: Here is the same case, I mentioned above.

**Review by Olexandr6 — CHANGES_REQUESTED** [review]
Your solution is good enough🔥, but let's work on some improvements

File: src/averageAges.js [hunk 2]
```diff
@@ -1,61 +1,63 @@
...
+
+/* Calculate Men Average Age */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let isMen = getSex(people, 'm');
```
Comment (jstmpelowycz) [inline src/averageAges.js:8]:
```suggestion
  let men = getBySex(people, 'm');
```

File: src/averageAges.js [hunk 3]
```diff
@@ -1,61 +1,63 @@
...
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const getSex = (people, sex) => people.filter(person => person.sex === sex);
```
Comment (jstmpelowycz) [inline src/averageAges.js:3]:
```suggestion
const getBySex = (people, sex) => people.filter(person => person.sex === sex);
```

File: src/averageAges.js [hunk 4]
```diff
@@ -1,61 +1,63 @@
...
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let isMen = getSex(people, 'm');
+
+  const isMenInCentury = isMen.filter(person =>
```
Comment (jstmpelowycz) [inline src/averageAges.js:10]:
```suggestion
  const menInCentury = ...
```

File: src/averageAges.js [hunk 5]
```diff
@@ -1,61 +1,63 @@
...
- * @return {number}
- */
+/* Calculate Women Average Age */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let isWoman = getSex(people, 'f');
+
+  const womenWithKids = people.filter(
+    person => people.some(child => child.mother === person.name));
```
Comment (jstmpelowycz) [inline src/averageAges.js:30]:
```suggestion
  const womenWithKids = people.filter(
    person => people.some(child => child.mother === person.name)
  );
```

File: src/averageAges.js [hunk 6]
```diff
@@ -1,61 +1,63 @@
...
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
+/* Calculate Men Average Difference */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let children = people.filter(
+    child => people.some(mother => mother.name === child.mother));
```
Comment (jstmpelowycz) [inline src/averageAges.js:47]:
```suggestion
  let children = people.filter(
    child => people.some(mother => mother.name === child.mother)
  );
```

File: src/averageAges.js (hunk 4 above)
Comment (jstmpelowycz) [reply src/averageAges.js:10]: you could follow this guideline:
boolean - for example, `isLoading`, `shouldUpdate`
single item – for example, `apple`, `box`, `student`
array of items – for example, `apples`, `boxes`, `studentsList`

File: src/averageAges.js [hunk 7]
```diff
@@ -1,61 +1,63 @@
...
+
+  isMen = century
+    ? isMenInCentury
+    : isMen;
+
+  const averageAgeMen = isMen.reduce(
+    (sum, man) => sum + (man.died - man.born), 0
+  ) / isMen.length;
+
+  return averageAgeMen;
```
Comment (jstmpelowycz) [inline src/averageAges.js:21]:
```suggestion
  return averageMenAge;
```

File: src/averageAges.js [hunk 8]
```diff
@@ -1,61 +1,63 @@
...
+  const isBoy = getSex(children, 'm');
+
+  children = onlyWithSon
+    ? isBoy
+    : children;
+
+  const averageAgeDiff = children.reduce(
+    (sum, child) => sum + (child.born - people.find(
+      mom => child.mother === mom.name).born), 0
+  ) / children.length;
```
Comment (jstmpelowycz) [inline src/averageAges.js:58]: that's unreadable, I'd something like that
```suggestion
  const averageAgeDiff = children.reduce((sum, child) => {
  	const mother = people.find(person => child.mother === person.name);
  	const ... = sum + (child.born - mother.born);
  	
  	return ...;
  }, 0) / children.length;
```

**Review by jstmpelowycz — APPROVED** [review]

**Pull request closed without merging**